### PR TITLE
[138] Move all stores into /stores directory

### LINKED ISSUE #138: move all stores into /stores/ dir
- [ ] add new stores dir for all stores
- [ ] update any needed imports
- [ ]  ensure all tests work.

### PR DESCRIPTION
**Pull request by Atul9**
##### Description
Moved all stores into `src/stores` directory

##### Checklist
- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Refers/Fixes
Fixes: #138

### CODE REVIEW

**Comment by eswarasai** [comment]
@Atul9 -- Thanks a lot for this PR and also for taking care of the issue for us. Can you please do a minor refactor by adding a `index.ts` under `src/stores` and exporting all the stores in the `index.ts`, so that it'd be easy to import across.

Example : 
`stores/index.ts` -->
```
export { BalanceAndAllowanceLazyStore } from './BalanceAndAllowanceLazyStore';
export { OrderFilledCancelledLazyStore } from './OrderFilledCancelledLazyStore';
```

`src/order_watcher/OrderStateWatcher.ts` -->
```
import { BalanceAndAllowanceLazyStore, OrderFilledCancelledLazyStore } from '../stores';
```

Makes sense?

**Comment by Atul9** [comment]
@eswarasai we are already exporting the stores in the respective file --
`export class BalanceAndAllowanceLazyStore` in  `stores/BalanceAndAllowanceLazyStore.ts`
why should we again export it from `index.ts`?

**Comment by eswarasai** [comment]
@Atul9 -- For the ease of imports throughout the library.

For example, if you have 5 stores which you need to import, you'd rather do -->

```
import { A, B, C, D, E } from 'store'
```

than -->

```
import A from 'store/A'
import B from 'store/B'
import C from 'store/C'
import D from 'store/D'
import E from 'store/E'
```

You can think of `store` as a module and it exports all it's internal sub-modules as part of it in `index.ts`. Hope this makes sense to you. Thanks!

**Comment by eswarasai** [comment]
@Atul9 -- Looks like the Travis build is failing. Can you please fix that and push the changes? In case you need help with this, feel free to join in our [Discord](https://marketprotocol.io/discord) channel #engineering to ask your queries. Thanks!

**Comment by Atul9** [comment]
@eswarasai done. I have updated the PR.

**Comment by eswarasai** [comment]
@Atul9 -- Can I know the reason why you've completely removed `src/index.ts` and moved everything inside to `src/stores/index.ts` alongside updating `rollup.config.js`?

**Comment by Atul9** [comment]
@eswarasai For your first comment on the PR, I thought it meant to move the `index.ts`. After going through it again I have updated as requested.

File: src/utilities/OrderStateUtils.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import BigNumber from 'bignumber.js';
 
-import { OrderFilledCancelledLazyStore } from '../OrderFilledCancelledLazyStore';
-import { BalanceAndAllowanceLazyStore } from '../BalanceAndAllowanceLazyStore';
+import { OrderFilledCancelledLazyStore } from '../stores/OrderFilledCancelledLazyStore';
```
Comment (eswarasai) [inline src/utilities/OrderStateUtils.ts]: @Atul9 -- You can shorten the imports here like below --

```
import {
  BalanceAndAllowanceLazyStore,
  OrderCollateralPoolAndTokenLazyStore,
  OrderFilledCancelledLazyStore
} from '../stores';
```

File: src/utilities/OrderStateUtils.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {
 import { Utils } from '../lib/Utils';
 import { RemainingFillableCalculator } from '../order_watcher/RemainingFillableCalc';
 import { Market } from '..';
-import { OrderCollateralPoolAndTokenLazyStore } from '../OrderCollateralPoolAndTokenLazyStore';
+import { OrderCollateralPoolAndTokenLazyStore } from '../stores/OrderCollateralPoolAndTokenLazyStore';
```
Comment (eswarasai) [inline src/utilities/OrderStateUtils.ts]: @Atul9 -- This import can also be shortened

```
import { OrderCollateralPoolAndTokenLazyStore } from '../stores';
```

File: test/BalanceAndAllowanceLazyStore.test.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import BigNumber from 'bignumber.js';
 
 import { Market, Utils } from '../src';
 
-import { BalanceAndAllowanceLazyStore } from '../src/BalanceAndAllowanceLazyStore';
+import { BalanceAndAllowanceLazyStore } from '../src/stores/BalanceAndAllowanceLazyStore';
```
Comment (eswarasai) [inline test/BalanceAndAllowanceLazyStore.test.ts]: Same as above

File: test/OrderCollateralPoolAndTokenLazyStore.test.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import BigNumber from 'bignumber.js';
 
 import { Market } from '../src';
 
-import { OrderCollateralPoolAndTokenLazyStore } from '../src/OrderCollateralPoolAndTokenLazyStore';
+import { OrderCollateralPoolAndTokenLazyStore } from '../src/stores/OrderCollateralPoolAndTokenLazyStore';
```
Comment (eswarasai) [inline test/OrderCollateralPoolAndTokenLazyStore.test.ts]: Same as above

File: test/OrderFilledCancelledStore.test.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { ERC20, MarketCollateralPool, MarketContract, SignedOrder } from '@marke
 import { Market, Utils } from '../src';
 import { constants } from '../src/constants';
 
-import { OrderFilledCancelledLazyStore } from '../src/OrderFilledCancelledLazyStore';
+import { OrderFilledCancelledLazyStore } from '../src/stores/OrderFilledCancelledLazyStore';
```
Comment (eswarasai) [inline test/OrderFilledCancelledStore.test.ts]: Same as above

**Comment by Atul9** [comment]
@eswarasai I have updated the PR.

**Comment by pelsasser** [comment]
Thanks @Atul9 for all the work and iterations here! this looks great.

**Pull request merged**